### DISCUSSION TITLE
Is the relationship between Mojo and Python is similar to TypeScript/Javascript or Kotlin/Java?

### DISCUSSION THREAD

**Discussion by bnyu** [modular/modular#68] [Q&A] [answered]
According README: _Mojo will be a superset of Python_. 
Does Mojo really gonna support all Python's now and future syntax?
Or just provide very well interaction between Mojo and Python?

**Comment by iamtimdavis** [comment, marked as answer]
The end goal of Mojo is indeed to be a superset of Python 🐍 We're not there today - but we are working extremely hard, along with our amazing Mojo🔥 community, to get there.

The CPython object has the same runtime representation, and uses the CPython interpreter which you can read about in the links below. All Mojo🔥 code is compiled code.

You can reference our docs:

[What kind of developers will benefit the most from Mojo?](https://docs.modular.com/mojo/faq.html#what-kind-of-developers-will-benefit-the-most-from-mojo)
[Is Mojo only for AI or can it be used for other stuff?](https://docs.modular.com/mojo/faq.html#is-mojo-only-for-ai-or-can-it-be-used-for-other-stuff)
[Why not enhance CPython (the major Python implementation) instead?](https://docs.modular.com/mojo/faq.html#why-not-enhance-cpython-the-major-python-implementation-instead)
